confluent_kafka/avro/CachedSchemaRegistryClient.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+#
+# derived from https://github.com/verisign/python-confluent-schemaregistry.git
+#
+import json
+import logging
+
+import requests
+from avro import schema
+
+from confluent_kafka.avro.serializer import Util
+from . import ClientError, VALID_LEVELS
+
+# Common accept header sent
+ACCEPT_HDR = "application/vnd.schemaregistry.v1+json, application/vnd.schemaregistry+json, application/json"
+
+
+class CachedSchemaRegistryClient(object):
+    """
+    A client that talks to a Schema Registry over HTTP
+
+    See http://confluent.io/docs/current/schema-registry/docs/intro.html
+
+    Errors communicating to the server will result in a ClientError being raised.
+
+    @:param: url: url to schema registry
+    """
+
+    def __init__(self, url, max_schemas_per_subject=1000):
+        """Construct a client by passing in the base URL of the schema registry server"""
+        self.log = logging.getLogger(__name__)
```

Comment:
nit: If you're going to have a logger in the class, you may consider using the class's fully qualified name.  Otherwise I don't know if there is an advantage with declaring it here as opposed to outside the class (e.g. `logger = logging.getLogger(__name__)`) because any class declared in this file would have the same logger name.